tests/test_robot_states.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+
+import omnigibson as og
+from omnigibson.macros import gm
+
+from omnigibson.sensors import VisionSensor
+
+def setup_environment(flatcache=True):
+    """
+    Sets up the environment with or without flatcache based on the flatcache parameter.
+    """
+    # Ensure any existing simulation is stopped
+    if og.sim is not None:
+        og.sim.stop()
+    
+    # Set global flags
+    gm.ENABLE_OBJECT_STATES = True
+    gm.USE_GPU_DYNAMICS = True
+    gm.ENABLE_FLATCACHE = flatcache  # Set based on function parameter
+    
+    # Define the environment configuration
+    config = {
+        "scene": {
+            "type": "Scene",
+        },
+        "robots": [
+            {
+                "type": "Fetch",
+                "obs_modalities": [],
+                "position": [150, 150, 100],
+                "orientation": [0, 0, 0, 1],
+                "controller_config": {
+                    "arm_0": {
+                        "name": "NullJointController",
+                        "motor_type": "position",
+                    },
+                },
+            }
+        ]
+    }
+
+    env = og.Environment(configs=config)
+    return env
+
+def camera_pose_test(flatcache):
+    env = setup_environment(flatcache)
+    robot = env.robots[0]
+    env.reset()
+    
+    # robot = og.sim.scene.object_registry("name", "robot0")
+    sensors = [s for s in robot.sensors.values() if isinstance(s, VisionSensor)]
+    assert len(sensors) > 0
+    vision_sensor = sensors[0]
+
+    # Get vision sensor world pose via directly calling get_position_orientation
+    sensor_world_pos, sensor_world_ori = vision_sensor.get_position_orientation()
+
+    sensor_world_pos_gt = np.array([150.16513062, 150.0, 101.36912537])
+    sensor_world_ori_gt = np.array([-0.29444987, 0.29444981, 0.64288363, -0.64288352])
+    
+    assert np.allclose(sensor_world_pos, sensor_world_pos_gt)
+    assert np.allclose(sensor_world_ori, sensor_world_ori_gt)
+    
+    # Now, we want to move the robot and check if the sensor pose is updated
+    old_camera_local_pose = vision_sensor.get_local_pose()
+    old_camera_world_pose = vision_sensor.get_position_orientation()
+    robot.set_position_orientation(position=[100, 100, 100])
+    new_camera_local_pose = vision_sensor.get_local_pose()
+    new_camera_world_pose = vision_sensor.get_position_orientation()
+    assert np.allclose(old_camera_local_pose[0], new_camera_local_pose[0])
+    assert not np.allclose(old_camera_world_pose[0], new_camera_world_pose[0])
```

Comment:
for all the `not np.all_close`, since we know the new pose we set, can we analytically compute the expected value and change them to `np.all_close(new_value)` instead?

Reply:
set parent scale for stress test